be/src/storage/persistent_index.cpp

```diff
@@ -718,16 +794,116 @@ class FixedMutableIndex : public MutableIndex {
+            // create a new empty _l0 file because all data in _l0 has write into _l1 files
+            std::string file_name = get_l0_index_file_name(_path, version);
+            WritableFileOptions wblock_opts;
+            wblock_opts.mode = FileSystem::CREATE_OR_OPEN_WITH_TRUNCATE;
+            ASSIGN_OR_RETURN(auto wfile, fs->new_writable_file(wblock_opts, file_name));
+            DeferOp close_block([&wfile] {
+                if (wfile) {
+                    wfile->close();
+                }
+            });
+            meta->clear_wals();
+            IndexSnapshotMetaPB* snapshot = meta->mutable_snapshot();
+            version.to_pb(snapshot->mutable_version());
+            PagePointerPB* data = snapshot->mutable_data();
+            // create a new empty _l0 file, set _offset to 0
+            data->set_offset(0);
+            data->set_size(0);
+            meta->set_format_version(PERSISTENT_INDEX_VERSION_1);
+            _offset = 0;
+            _page_size = 0;
+            break;
+        }
+        case kSnapshot: {
+            std::string file_name = get_l0_index_file_name(_path, version);
+            // be maybe crash after create index file during last commit
+            // so we delete expired index file first to make sure no garbage left
+            FileSystem::Default()->delete_file(file_name);
+            size_t snapshot_size = dump_bound();
+            phmap::BinaryOutputArchive ar_out(file_name.data());
+            if (!dump(ar_out)) {
+                std::string err_msg = strings::Substitute("faile to dump snapshot to file $0", file_name);
+                LOG(WARNING) << err_msg;
+                return Status::InternalError(err_msg);
+            }
+            // dump snapshot success, set _index_file to new snapshot file
+            WritableFileOptions wblock_opts;
+            wblock_opts.mode = FileSystem::MUST_EXIST;
+            ASSIGN_OR_RETURN(_index_file, fs->new_writable_file(wblock_opts, file_name));
+            meta->clear_wals();
+            IndexSnapshotMetaPB* snapshot = meta->mutable_snapshot();
+            version.to_pb(snapshot->mutable_version());
+            PagePointerPB* data = snapshot->mutable_data();
+            data->set_offset(0);
+            data->set_size(snapshot_size);
+            meta->set_format_version(PERSISTENT_INDEX_VERSION_1);
+            _offset = snapshot_size;
+            _page_size = 0;
+            break;
+        }
+        case kAppendWAL: {
+            IndexWalMetaPB* wal_pb = meta->add_wals();
+            version.to_pb(wal_pb->mutable_version());
+            PagePointerPB* data = wal_pb->mutable_data();
+            data->set_offset(_offset);
+            data->set_size(_page_size);
+            meta->set_format_version(PERSISTENT_INDEX_VERSION_1);
+            _offset += _page_size;
+            _page_size = 0;
+            break;
+        }
+        default: {
+            return Status::InternalError("Unknown commit type");
```

Comment:
Add tablet_id into the error log.

Reply:
> Add tablet_id into the error log.

The tablet_id is not available here. More detailed error messages will be printed at the upper level call